lib/carin_for_blue_button_test_kit/carin_search_test.rb

```diff
@@ -199,67 +200,38 @@ def run_include_search(search_params, param_value)
         if param_value != 'ExplanationOfBenefit:*'
           paths.each do |path|
               values_found = resolve_path(resource, path)
-
-              values_found.each do |reference|
-                reference_found = find_included_resource(reference, returned_resources_all)
-
-                # If at least one reference is not found, set reference_bool to false and do not change back to true for any other found references
-                if !reference_found
-                    reference_bool = false
-                end
-              end
-
               match_found = (values_found.length > 0)
 
               break if match_found
           end
           assert match_found, "Returned resource did not match the search parameter"
-          assert reference_bool, "Returned resource did not include the _include resource parameter"
-          return
         else
-          values_found = []
-          reference_bool = true
-
           paths.each do |path|
               values_found += resolve_path(resource, path)
           end
-
-          values_found.each do |reference|
-            reference_found = find_included_resource(reference, returned_resources_all)
-
-            # If at least one reference is not found, set reference_bool to false and do not change back to true for any other found references
-            if !reference_found
-                reference_bool = false
-            end
-          end
-
           match_found = (values_found.length >= 5)
+          assert match_found, "Returned resource did not match the search parameter"  
+        end  
+      end  
 
-          assert match_found, "Returned resource did not match the search parameter"
-          assert reference_bool, "Returned resource did not include the _include resource parameter"
+      referenced_resource_types = values_found.map{ |reference| reference.reference.split('/')[-2]}.uniq
```

Comment:
This may not work. If the reference is "http://example.com/Patient/abc/_history/2" then `reference.reference.split('/')[-2]` returns "_history".

A simple solution is to chang the `include_param_paths` method to return:

```
[
  {
    "path": "patient",
    "targetType": ["Patient"]
  }
]
```

Then, you don't need this line and you can change next line to 

A more complicated solution is change the generator to generate this returned structure in the test module, i.e. `explanation_of_benefit_explanation_of_benefit_patient_search.rb` and send in as the 3rd input parameter into this method.

My suggestion: Implement the simple soution here, add a new ticket to update generator.